content/client-lib-development-guide/features.textile

```diff
@@ -232,6 +232,22 @@ h3(#rest-auth). Auth
 ** @(RSA16c)@ Is set with the current token (if applicable) on instantiation and each time it is replaced, whether the result of an explicit @Auth#authorize@ operation, or a library-initiated renewal resulting from expiry or a token error response
 ** @(RSA16d)@ Is empty if there is no current token, including after a previous token has been determined to be invalid or expired, or if the library is using basic auth
+* @(RSA17)@ @Auth#revokeTokens@ function: This function is intended to be used to revoke tokens using provided @TokenRevocationRequest@ details. It internally creates request for  "REST revokeTokens endpoint":/rest-api/#revokeTokens and handles the response.
+** @(RSA17a)@ Method signature is @revokeTokens(TokenRevocationRequest)@.
+** @(RSA17b)@ If the @TokenRevocationRequest#key@ attribute is not provided, then the function must throw an error with a @"Key is not provided"@ message.
+** @(RSA17c)@ If the @TokenRevocationRequest#targets@ attribute is null or invalid (as per @TF3@ ), then the function must throw an error with a @"targets are invalid"@ message.
+** @(RSA17d)@ If the @TokenRevocationRequest#key@ is provided and @TokenRevocationRequest#targets@ valid, then the function must create the request and execute it.
+** @(RSA17e)@ Function returns a @TokenRevocationResponse@ object that contains the result of the request if request is successful.
+** @(RSA17f)@ Function throws an appropriate error/exception  if the request fails either for the reasons of invalid request, or as  a result of request failing.
+** @(RSA17g)@ Tests in this section must exist in client library.
+*** @(RSA17g1)@ Given a null @TokenRevocationRequest#key@ attribute, verify that the function throws an error with a @"Key is not provided"@ message.
+*** @(RSA17g2)@ Given a null @TokenRevocationRequest#targets@ attribute, verify that the function throws an error with a @"targets are invalid"@ message.
+*** @(RSA17g3)@ Given an empty @TokenRevocationRequest#targets@ attribute, verify that the function throws an error with a @"targets are invalid"@ message.
+*** @(RSA17g4)@ Given a non-null @TokenRevocationRequest#key@ attribute and a valid @TokenRevocationRequest#targets@ attribute, verify that the function creates the request and executes it.
+*** @(RSA17g5)@ Given a non-null but invalid @TokenRevocationRequest#key@ attribute and a valid @TokenRevocationRequest#targets@ attribute, verify that client library throws an API error.
+*** @(RSA17g6)@ Given a valid @TokenRevocationRequest#key@ attribute and a valid @TokenRevocationRequest#targets@ attribute, verify that the function executes the request and returns a @TokenRevocationResponse@ with targets matching with the original targets in request.
+*** @(RSA17g7)@ Given valid required attributes in @TokenRevocationRequest@ and a timestamp in the future for @TokenRevocationRequest#issuedBefore@ attribute, verify that the request is rejected. 
```

Comment:
This is not clear, does this rely on the local clock only or rely on the server clock only?  I would advocate for simplicity and rely on the server clock as no clock sync is needed then.

Reply:
No, what I am saying is that the client does not need to enforce this, this creates extra code in the SDK, a dependency on time sync, and what gain do we get from this?  The client will get an error code anyway.  We should be explicit

Reply:
Because `issuedBefore` is a timestamp that can (optionally) be provided by client, I thought it would be a good way to verify client SDK handles passing this correctly to the REST API. It wasn't to verify  the API behaviour.

Reply:
Removed test cases relating to `issuedBefore` 4207e8ec4d0d7483e64df76ec287741ba43ec719